ansible/roles/test/files/ptftests/py3/advanced-reboot.py

```diff
@@ -1848,25 +1888,50 @@ def sniff_in_background(self, wait=None):
 
     def tcpdump_sniff(self, wait=300, sniff_filter=''):
         """
-        @summary: PTF runner -  runs a sniffer in PTF container.
+        @summary: PTF runner -  runs a sniffer in vmhost(server) or the PTF container.
         Args:
             wait (int): Duration in seconds to sniff the traffic
             sniff_filter (str): Filter that tcpdump will use to collect only relevant packets
         """
         try:
-            capture_pcap = ("/tmp/capture_%s.pcap" % self.logfile_suffix
-                            if self.logfile_suffix is not None else "/tmp/capture.pcap")
-            subprocess.call(["rm", "-rf", capture_pcap])  # remove old capture
+            subprocess.call(["rm", "-rf", self.capture_pcap])
             self.kill_sniffer = False
-            self.start_sniffer(capture_pcap, sniff_filter, wait)
-            self.create_single_pcap(capture_pcap)
-            self.packets = scapyall.rdpcap(capture_pcap)
+
+            if self.vmhost_external_port:
+                self.start_sniffer_on_vmhost(self.remote_capture_pcap, sniff_filter, wait)
+                self.vmhost_connection.fetch(self.remote_capture_pcap, self.capture_pcap)
+            else:
+                self.start_sniffer_on_ptf(self.capture_pcap, sniff_filter, wait)
+                self.create_single_pcap(self.capture_pcap)
+
+            self.packets = scapyall.rdpcap(self.capture_pcap)
             self.log("Number of all packets captured: {}".format(len(self.packets)))
         except Exception:
             traceback_msg = traceback.format_exc()
             self.log("Error in tcpdump_sniff: {}".format(traceback_msg))
 
-    def start_sniffer(self, pcap_path, tcpdump_filter, timeout):
+    def start_sniffer_on_vmhost(self, pcap_path, tcpdump_filter, timeout):
+        """
+        Start tcpdump sniffer on all data interfaces, and kill them after a specified timeout
+        """
+        interface = self.test_params['vmhost_external_port']
+        cmd = f"sudo nohup tcpdump -i {interface} {tcpdump_filter} -w {pcap_path}"
```

Comment:
If the physical interface here is handling traffic for many testbeds (meaning combinations of physical devices, PTF containers, and neighbor containers/VMs), then this is going to be a lot of traffc, which means analyzing the traffic and filtering to what is actually needed is going to be a lot harder.

Reply:
As far as I know, one logic vm_host should be used only for one testbed, right? And even if the physical server is shared by multiple testbeds, the physical interface of the vm_host is only used by one single vm_host and testbed, right?
And because we have tcpdump_filter here, which is "tcp and tcp dst port 5000 and tcp src port 1234 and not icmp", only the traffic for the test will be captured. Actually the pcap is the same as before because we have this filter.

Reply:
No. One physical server with one physical network interface may be used by multiple testbeds. The traffic for each port for each testbed might be in a separate VLAN on that physical interface. This is, in fact, the setup that we use for our testbed environment.

While this approach might work in certain setups, it is not generic, and breaks the PTF "encapsulation".

Reply:
In that case, I can change it to use the original capture in ptf by default, and use the physical port only when the user explicitly specify it with a pytest parameter. Do you think it will work for you?

Reply:
@saiarcot895 Could you please check and reply?